### DISCUSSION TITLE
Any way to setup Lazy keymaps instead of plugin mappings, (or somehow get whichkey to display the plugin mappings)?

### DISCUSSION THREAD

**Discussion by jamesonBradfield** [kylechui/nvim-surround#320] [Q&A]
Just as the title says, I have a hard time remembering keybinds and whichkey is a lifesaver.

**Comment by kylechui** [comment]
I would imagine which-key should automatically register nvim-surround keymaps; try hitting `y` in normal mode and see if the `ys` map pops up.

**Comment by dgokcin** [comment]
@jamesonBradfield @kylechui did you guys figure out a way to display the keymaps with which key? it is really hard to remember them all and I haven't been able to enable the mappings with the following config.

```lua
return {
  "kylechui/nvim-surround",
  version = "*",
  event = "BufRead",
  vscode = "true",
  config = function()
    require("nvim-surround").setup({})
    if not vim.g.vscode then
      -- Setup which-key descriptions for nvim-surround keymaps
      local wk = require("which-key")
      local mappings = {
        { mode = { "n", "v" } },
        { "ys", group = "surround" },
        { "ysa", group = "around" },
        { 'ysa"', desc = "Surround around double quotes" },
        { "ysa'", desc = "Surround around single quotes" },
        { "ysa(", desc = "Surround around parentheses" },
        { "ysa<", desc = "Surround around angle brackets" },
        { "ysaW", desc = "Surround around WORD" },
        { "ysa[", desc = "Surround around brackets" },
        { "ysa`", desc = "Surround around backticks" },
        { "ysap", desc = "Surround around paragraph" },
        { "ysas", desc = "Surround around sentence" },
        { "ysat", desc = "Surround around tag block" },
        { "ysaw", desc = "Surround around word" },
        { "ysa{", desc = "Surround around braces" },
        { "ysi", group = "inside" },
        { 'ysi"', desc = "Surround inside double quotes" },
        { "ysi'", desc = "Surround inside single quotes" },
        { "ysi(", desc = "Surround inside parentheses" },
        { "ysi<", desc = "Surround inside angle brackets" },
        { "ysiW", desc = "Surround inside WORD" },
        { "ysi[", desc = "Surround inside brackets" },
        { "ysi`", desc = "Surround inside backticks" },
        { "ysip", desc = "Surround inside paragraph" },
        { "ysis", desc = "Surround inside sentence" },
        { "ysit", desc = "Surround inside tag block" },
        { "ysiw", desc = "Surround inside word" },
        { "ysi{", desc = "Surround inside braces" },
      }
      wk.add(mappings)
    end
  end,
}

```

**Reply by kylechui** [reply]
The issue is that `ys` is the actual keymap being defined, as `a"` (or equivalent) is just a motion that is being used. There's no `ysa"` keymao being defined. I would advise you to read `:h motion` or :h a"` to learn more about how they work.

**Reply by jamesonBradfield** [reply]
Yeah, I was completely wrong, there is no need to map them like that, that being said i find the default descriptions pretty verbose.
I believe the mappings are already added by the plugin to whichkey and (with the architecture needed for such a change to make the plugin favor one plugin manager vs another would be odd).

**Reply by jamesonBradfield** [reply]
on a side tangent, kylechui if I wanted to get hands on experience with some plugin dev aside from my own, do you know of any issues that would be easy for a beginner to solve? Or can you reccomend a strategy in finding a way I can contribute?

**Reply by kylechui** [reply]
To be honest I've not really been working on this plugin much aside from answering questions and maintaining it (e.g. fixing bugs), so I'm not quite sure this would be the best place to really get more experience developing in other people's plugins. You can take a look through the bugs in the issues tab, but I think most of them are there because I can't solve them (in a nice way :sweat_smile:). I've grown pretty averse to changing the plugin in any meaningful way in order to prevent any new breaking changes from occurring as well
